src/kontrolilo/yarn.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+import csv
+from subprocess import run
+from typing import List
+
+from kontrolilo.base_checker import BaseLicenseChecker
+from kontrolilo.shared_main import shared_main
+from kontrolilo.configuration.package import Package
+
+
+class YarnLicenseChecker(BaseLicenseChecker):
+    def prepare_directory(self, directory: str):
+        run('yarn install', capture_output=True, check=True, cwd=directory, shell=True)
```

Comment:
Except for this line, the npm and yarn versions are exactly the same. I would suggest introducing a superclass that contains the rest of the code and leaves this detail to the specific version.